src/App.tsx

```diff
@@ -10,13 +10,18 @@ import Register from "./pages/Register";
 import { ProtectedRoute } from "./components/Auth/ProtectedRoute";
 import PageLoader from "./pages/PageLoader";
 import Profile from "./pages/Profile";
-import { useUserContext } from "./context/UserContext";
 import "./App.css";
+import { RootState } from "./state/store";
+import { useDispatch, useSelector } from "react-redux";
+import { setUser } from "./state/slices/userSlice";
 import { PostContextProvider } from "./context/PostContext";
 
 function App() {
   const { isLoading, isAuthenticated, getAccessTokenSilently } = useAuth0();
-  const { user, setUser } = useUserContext();
+  // const { user, setUser } = useUserContext();
+
+  const user = useSelector((state: RootState) => state.user.value);
+  const dispatch = useDispatch();
 
   useEffect(() => {
```

Comment:
We should look into if we can use `createAsyncThink` to do this initial fetch (which comes installed with Redux Toolkit)

Reply:
We'll also handle this in another ticket. Looks like `createAsyncThunk` just adds some syntax sugar to make things more neat